lib/pbench/cli/agent/commands/tools/list.py

```diff
@@ -36,35 +45,45 @@ def execute(self):
             host_tools = {}
             for group in groups:
                 host_tools[group] = {}
-                for path in self.gen_tools_group_dir(group).glob("*/**"):
-                    if self.context.with_option:
-                        host_tools[group][path.name] = [
-                            p.read_text() for p in self.tools(path)
-                        ]
-                    else:
-                        host_tools[group][path.name] = [p for p in self.tools(path)]
+                try:
+                    for path in self.gen_tools_group_dir(group).glob("*/**"):
+                        if self.context.with_option:
+                            host_tools[group][path.name] = [
+                                p.read_text() for p in self.tools(path)
+                            ]
+                        else:
+                            host_tools[group][path.name] = [p.name for p in self.tools(path)]
+                except BadToolGroup:
+                    self.logger.error("Bad tool group name: %s", group)
+                    return 1
             if host_tools:
                 for k, v in host_tools.items():
                     for h, t in v.items():
-                        print("%s: %s %s" % (k, h, t))
+                        print(f"{k}: {h} {t}")
         else:
             # List the groups which include this tool
             group_list = []
             for group in groups:
-                tg_dir = self.gen_tools_group_dir(group)
-                if not tg_dir.exists():
-                    self.logger.error("bad or missing tool group %s", group)
+                try:
+                    tg_dir = self.gen_tools_group_dir(group)
+                    if not tg_dir.exists():
+                        self.logger.error("bad or missing tool group %s", group)
+                        status = 1
+                        continue
+                except BadToolGroup:
+                    self.logger.error("Bad Tool Group: %s", group)
+                    status = 1
                     continue
-
                 for path in tg_dir.iterdir():
                     # Check to see if the tool is in any of the hosts.
-                    if self.context.name in self.tools(path):
+                    if self.context.name in [tool.name for tool in self.tools(path)]:
                         group_list.append(group)
             if group_list:
-                print(
-                    "tool name: %s groups: %s"
-                    % (self.context.name, ", ".join(group_list))
-                )
+                print(f"tool name: {self.context.name} groups: {', '.join(group_list)}")
+            elif status == 0:
+                # the group list is empty but not because of a bad tool group, so set an "other"
+                status = 2
```

Comment:
Does this case just mean the specified tool is not found in any group?  The previous code seems to exit successfully on this case.

Reply:
Correct - and it's true the previous code exited successfully, but should it?